gateway-sp-comms/src/single_sp/update.rs

```diff
@@ -257,57 +258,137 @@ fn read_auxi_check_from_tlvc(data: &[u8]) -> Result<[u8; 32], UpdateError> {
     })
 }
 
+/// Isolate extraction of bootleby from old-format archives.
+/// TODO: Purge old-format archives and delete this code.
+fn bootleby_from_old_style_archive(
+    image: Vec<u8>,
+    log: &Logger,
+) -> Result<Vec<u8>, UpdateError> {
+    // Try the pre-v1.2.0 Bootleby archive format.
+    let cursor = Cursor::new(image.as_slice());
+    let mut archive = zip::ZipArchive::new(cursor).map_err(|zip_error| {
+        // Return the original Hubris Archive error instead
+        // of our attempted zip extraction error.
+        HubtoolsError::ZipError(zip_error)
+    })?;
+
+    let mut rot_image = vec![];
+    for i in 0..archive.len() {
+        let mut file = match archive.by_index(i) {
+            Ok(file) => file,
+            Err(e) => {
+                error!(
+                    log,
+                    "did not find bootleby.bin in old-style archive: {e}"
+                );
+                return Err(UpdateError::InvalidArchive);
+            }
+        };
+        if file.name() == "bootleby.bin" {
+            if file.read_to_end(&mut rot_image).is_err() {
+                error!(log, "invalid archive");
+                return Err(UpdateError::InvalidArchive);
+            }
+            debug!(log, "using bootleby.bin from old-style archive");
+            break;
+        }
+    }
+    Ok(rot_image)
```

Comment:
This doesn't seem right if we don't find a `bootleby.bin` - it will return an empty vec, right? Could we change the success case in the loop to `return Ok(rot_image)` instead of `break`, then return an error here instead?

Reply:
Good catch.